bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudAccountHandler.java

```diff
@@ -0,0 +1,222 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.melcloud.internal.handler;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.melcloud.internal.api.MelCloudConnection;
+import org.openhab.binding.melcloud.internal.api.json.Device;
+import org.openhab.binding.melcloud.internal.api.json.DeviceStatus;
+import org.openhab.binding.melcloud.internal.api.json.HeatpumpDeviceStatus;
+import org.openhab.binding.melcloud.internal.config.AccountConfig;
+import org.openhab.binding.melcloud.internal.discovery.MelCloudDiscoveryService;
+import org.openhab.binding.melcloud.internal.exceptions.MelCloudCommException;
+import org.openhab.binding.melcloud.internal.exceptions.MelCloudLoginException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link MelCloudAccountHandler} is the handler for MELCloud API and connects it
+ * to the webservice.
+ *
+ * @author Luca Calcaterra - Initial contribution
+ * @author Pauli Anttila - Refactoring
+ * @author Wietse van Buitenen - Return all devices, added heatpump device
+ */
+public class MelCloudAccountHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(MelCloudAccountHandler.class);
+
+    private MelCloudConnection connection;
+    private List<Device> devices;
+    private ScheduledFuture<?> connectionCheckTask;
+    private AccountConfig config;
+    private boolean loginCredentialError;
+
+    public MelCloudAccountHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(MelCloudDiscoveryService.class);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing MELCloud account handler.");
+        config = getConfigAs(AccountConfig.class);
+        connection = new MelCloudConnection();
+        devices = Collections.emptyList();
+        loginCredentialError = false;
+        startConnectionCheck();
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Running dispose()");
+        stopConnectionCheck();
+        connection = null;
+        devices = Collections.emptyList();
+        config = null;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
```

Comment:
Can you handle the REFRESH command in a sensible way? If it's possible please do so.

Reply:
`MelCloudAccountHandler` doesn't update any values on its own, `scheduler` does check and maintain connection at fixed interval of 60 seconds.
Not sure if refresh would add benefit in this case, I think this is why was chosen not to handle this command.